### ISSUE TITLE
Add `aria-*` and `role` support to TUI components

### ISSUE THREAD

**Issue by webbower** [thumbtack/thumbprint#405]
The current HTML attribute whitelist on TUI components prevents setting accessibility attributes on them. `role` and `aria-*` attributes should be added to the whitelist. The Link and Button components are high on the wish list for these.

The inspiring use case is expand/collapse functionality a la http://heydonworks.com/practical_aria_examples/#progressive-collapsibles which is using `TextButton`
